main.js

```diff
@@ -257,13 +303,44 @@
 				console.log(dataSource.id, data);
                 const records = data.result.records;
 
+                var filterContainer = document.createElement("div");
+                filterContainer.className = "checkBtn";
+
+                var filter = document.createElement("input");
+                filter.type = "checkbox";
+                filter.id = dataSourceName.toLowerCase() + "Check";
+                filter.checked = true;
+
+                var filterLabel = document.createElement("label");
+                filterLabel.htmlFor = dataSourceName.toLowerCase() + "Check";
+                filterLabel.innerHTML = dataSource.icon.options.html;
```

Comment:
We should also have the name of the dataset (just `dataSourceName.toLowerCase()`)